### PR TITLE
Ammo box changes

### PR DESCRIPTION
**Pull request by JeserTheFox**
## About The Pull Request

1. Ammo boxes now support stacking! Compact storing is best storing! 2 per tile for regular ammo boxes, 4 per tile for misc ammo boxes (nail ammo ones have no stacking because of snowflake sprite, god I hate it so much).
2. Reworked explosion when flaming ammo boxes. Boxes explosion now scales with amount of rounds inside. Our very safe cardboard ammo boxes prevent explosion if amount of rounds inside does not hit a certain threshold. (Eliminates the use of rounds boxes and 1 round inside as improvised grenades). Fully loaded boxes now explode harder than before, however.
3. Boxes now have burning time, which is reduced depending on amount of bullets inside. If box is going to blow up, it gets different overlay on it.
4. New sprites: "burning" and "burning going to explode" overlays animations, new marking overlay for flare box to be easily identifiable.

Regular boxes stacking:
dreamseeker_6t3LFPkbJ0
Miscellaneous supplies boxes stacking:
dreamseeker_mNl5kkJNki
Lower than explosion threshold ammo amount box on fire:
dreamseeker_S6XoUWCIgY
Flare box on fire:
dreamseeker_W9R6QaHuYZ
Full ammo boxes on fire:
dreamseeker_fJC6DrIGHG

## Why It's Good For The Game

Fixes exploitable mechanic, more QoL, more cool animations and explosions!

## Changelog

:cl: Jeser
qol: Supply and logistics workers, rejoice! Ammo boxes became stackable. Regular ammo boxes can be stacked up to 2 per tile, miscellaneous supply boxes (flares and MREs, currently) up to 4 per tile. No stacking for nail gun ammo, because of a snowflake sprite.
qol: Added flare box overlay to make it easier to separate MRE from flare boxes.
balance: Reworked how explosion happens when flaming ammo boxes. 
code: Explosion level scales with amount of rounds inside. Boxes will not explode until amount of rounds inside won't reach a certain threshold. Fully filled boxes explode harder now. 
code: After being flamed, boxes now spend some time being on fire before they will be deleted or explode, during which they receive a new on fire overlay. This time is also based on amount of ammo in the box. If the box is going to blow up, overlay will be slightly different to indicate that.
imageadd: New sprites include burning and burning, going to explode overlays as well as M94 flare box markings overlay.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/ammunition.dm [hunk 1]
```diff
@@ -148,12 +147,17 @@ They're all essentially identical when it comes to getting the job done.
 	gun_type = source.gun_type
 
 //~Art interjecting here for explosion when using flamer procs.
-/obj/item/ammo_magazine/flamer_fire_act(damage, flame_cause_data)
-	switch(current_rounds)
-		if(0) return
-		if(1 to 100) explosion(loc,  -1, -1, 0, 2, , , , flame_cause_data) //blow it up.
-		else explosion(loc,  -1, -1, 1, 2, , , , flame_cause_data) //blow it up HARDER
-	qdel(src)
+/obj/item/ammo_magazine/flamer_fire_act(var/damage, var/datum/cause_data/flame_cause_data)
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:150]: flamer fuel tanks should not explode

File: code/modules/projectiles/ammunition.dm [hunk 2]
```diff
@@ -315,6 +319,40 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
+/obj/item/ammo_box/flamer_fire_act(var/severity, var/datum/cause_data/flame_cause_data)
+	if(burning)
+		return
+	burning = TRUE
+
+	SetLuminosity(3)
+	apply_fire_overlay()
+	addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, src), 5 SECONDS)
+
+/obj/item/ammo_box/proc/apply_fire_overlay(var/will_explode = FALSE)
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:342]: maybe this should change colour with the colour of the fire like how burning mobs do?

File: code/modules/projectiles/ammunition.dm [hunk 3]
```diff
@@ -315,6 +319,40 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
+	//original fire overlay is made for standard mag boxes, so they don't need additional offsetting
+	var/offset_x = pixel_x
+	var/offset_y = pixel_y
+	if(limit_per_tile == 4)	//misc boxes (mre, flares etc)
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:346]: there has got to be a better way

use subtyping procs etc

File: code/modules/projectiles/ammunition.dm [hunk 4]
```diff
@@ -315,6 +319,40 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
+		offset_x += 1
+		offset_y += -6
+	else if(istype(src, /obj/item/ammo_box/magazine/smg/nailgun))	//this snowflake again
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:349]: as above

File: code/modules/projectiles/ammunition.dm [hunk 5]
```diff
@@ -398,27 +447,110 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
+				severity += AM.current_rounds
+			severity = round(severity / 150)
+		//we need a lot of bullets to produce an explosion. Different scaling because of very different ammo counts
+
+		if(severity > 0)
+			if(istype(loc, /obj/structure/magazine_box))
+				var/obj/structure/magazine_box/host_box = loc
+				host_box.apply_fire_overlay(TRUE)
+				host_box.SetLuminosity(3)
+				host_box.visible_message(SPAN_WARNING("\The [src] catches on fire and ammunition starts cooking off! Its gonna blow!"))
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:513]:
```suggestion
				host_box.visible_message(SPAN_WARNING("\The [src] catches on fire and ammunition starts cooking off! It's gonna blow!"))
```

File: code/modules/projectiles/ammunition.dm [hunk 6]
```diff
@@ -398,27 +447,110 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
+			else
+				apply_fire_overlay(TRUE)
+				SetLuminosity(3)
+				visible_message(SPAN_WARNING("\The [src] catches on fire and ammunition starts cooking off! Its gonna blow!"))
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:517]:
```suggestion
				visible_message(SPAN_WARNING("\The [src] catches on fire and ammunition starts cooking off! It's gonna blow!"))
```

File: code/modules/projectiles/ammunition.dm [hunk 7]
```diff
@@ -398,27 +447,110 @@ Turn() or Shift() as there is virtually no overhead. ~N
 	new /obj/item/stack/sheet/cardboard(T)
 	qdel(src)
 
-/obj/item/ammo_box/magazine/proc/deploy_ammo_box(mob/living/user, turf/T)
-	for(var/obj/structure/magazine_box/MB in T.contents)
-		to_chat(user, SPAN_WARNING("There is a [MB] deployed here already."))
+/obj/item/ammo_box/magazine/proc/deploy_ammo_box(var/mob/living/user, var/turf/T)
+	if(burning)
+		to_chat(user, SPAN_DANGER("It's on fire and can explode!"))
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:452]:
```suggestion
		to_chat(user, SPAN_DANGER("It's on fire and might explode!"))
```

File: code/modules/projectiles/ammunition.dm [hunk 8]
```diff
@@ -379,9 +423,14 @@ Turn() or Shift() as there is virtually no overhead. ~N
 				to_chat(user, SPAN_INFO("It feels about half full."))
 				return
 			to_chat(user, SPAN_INFO("It feels almost full."))
+	if(burning)
+		to_chat(user, SPAN_DANGER("It's on fire and can explode!"))
 
 /obj/item/ammo_box/magazine/attack_self(mob/living/user)
 	..()
+	if(burning)
+		to_chat(user, SPAN_DANGER("It's on fire and can explode!"))
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:432]:
```suggestion
		to_chat(user, SPAN_DANGER("It's on fire and might explode!"))
```

File: code/modules/projectiles/ammunition.dm [hunk 9]
```diff
@@ -379,9 +423,14 @@ Turn() or Shift() as there is virtually no overhead. ~N
 				to_chat(user, SPAN_INFO("It feels about half full."))
 				return
 			to_chat(user, SPAN_INFO("It feels almost full."))
+	if(burning)
+		to_chat(user, SPAN_DANGER("It's on fire and can explode!"))
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:427]:
```suggestion
		to_chat(user, SPAN_DANGER("It's on fire and might explode!"))
```

File: code/modules/projectiles/ammunition.dm [hunk 10]
```diff
@@ -398,27 +447,110 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
-		to_chat(user, SPAN_WARNING("There is a [MB] deployed here already."))
+/obj/item/ammo_box/magazine/proc/deploy_ammo_box(var/mob/living/user, var/turf/T)
+	if(burning)
+		to_chat(user, SPAN_DANGER("It's on fire and can explode!"))
 		return
+
+	var/box_on_tile = 0
+	for(var/obj/structure/magazine_box/found_MB in T.contents)
+		if(limit_per_tile != found_MB.limit_per_tile)
+			to_chat(user, SPAN_WARNING("You can't deploy different size boxes in one place."))
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:458]:
```suggestion
			to_chat(user, SPAN_WARNING("You can't deploy different size boxes in one place!"))
```

File: code/modules/projectiles/ammunition.dm [hunk 11]
```diff
@@ -398,27 +447,110 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
+			return
+		box_on_tile++
+		if(box_on_tile >= limit_per_tile)
+			to_chat(user, SPAN_WARNING("There is a maximum possible amount of boxes deployed here already."))
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:462]:
```suggestion
			to_chat(user, SPAN_WARNING("You can't cram any more boxes in here!"))
```

File: code/modules/projectiles/ammunition.dm [hunk 12]
```diff
@@ -398,27 +447,110 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
 	if(!proximity)
 		return
 	if(isturf(target))
 		var/turf/T = target
 		if(!T.density)
 			deploy_ammo_box(user, T)
 
+//---------------------FIRE HANDLING PROCS
+
+/obj/item/ammo_box/magazine/flamer_fire_act(var/damage, var/datum/cause_data/flame_cause_data)
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:491]: this is full of duped code from /ammo_box/flamer_fire_act() and needs to be separated out into procs on /ammo_box (such as for getting severity, and for igniting)

File: code/modules/projectiles/ammunition.dm [hunk 13]
```diff
@@ -398,27 +447,110 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
+		//need to make sure we also delete the structure box
+		addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, host_box), 5 SECONDS)
+	else
+		visible_message(SPAN_WARNING("\The [src] catches on fire!"))
+		apply_fire_overlay()
+		SetLuminosity(3)
+		addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, src), 5 SECONDS)
+	return
+
+/obj/item/ammo_box/magazine/apply_fire_overlay(var/will_explode = FALSE)
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:535]: why more duped code

File: code/modules/projectiles/ammunition.dm [hunk 14]
```diff
@@ -905,8 +1041,63 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
+		pixel_y = possible_offsets[available_offset][2]
+		return
+
+//---------------------INTERACTION PROCS
+
+/obj/structure/magazine_box/MouseDrop(over_object, src_location, over_location)
+	..()
+	if(over_object == usr && Adjacent(usr))
+		if(burning)
+			to_chat(usr, SPAN_DANGER("It's on fire and can explode!"))
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:1072]:
```suggestion
			to_chat(usr, SPAN_DANGER("It's on fire and might explode!"))
```

File: code/modules/projectiles/ammunition.dm [hunk 15]
```diff
@@ -905,8 +1041,63 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
+			return
+
+		if(!ishuman(usr))
+			return
+		visible_message(SPAN_NOTICE("[usr] picks up [name]."))
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:1077]:
```suggestion
		visible_message(SPAN_NOTICE("[usr] picks up the [name]."))
```

File: code/modules/projectiles/ammunition.dm [hunk 16]
```diff
@@ -905,8 +1041,63 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
+		return
+	to_chat(user, SPAN_INFO("[SPAN_HELPFUL("Click")] on the box with an empty hand to take a magazine out. [SPAN_HELPFUL("Drag")] it onto yourself to pick it up."))
+	if(item_box.handfuls)
+		var/obj/item/ammo_magazine/AM = locate(/obj/item/ammo_magazine) in item_box.contents
+		if(AM)
+			to_chat(user, SPAN_INFO("It has roughly [round(AM.current_rounds/5)] handfuls remaining."))
+	else
+		to_chat(user, SPAN_INFO("It has [item_box.contents.len] magazines out of [item_box.num_of_magazines]."))
+	if(burning)
+		to_chat(user, SPAN_DANGER("It's on fire and can explode!"))
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:1095]:
```suggestion
		to_chat(user, SPAN_DANGER("It's on fire and might explode!"))
```

File: code/modules/projectiles/ammunition.dm [hunk 17]
```diff
@@ -905,8 +1041,63 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
 
 /obj/structure/magazine_box/attack_hand(mob/living/user)
+	if(burning)
+		to_chat(user, SPAN_DANGER("It's on fire and can explode!"))
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:1099]:
```suggestion
		to_chat(user, SPAN_DANGER("It's on fire and might explode!"))
```

File: code/modules/projectiles/ammunition.dm [hunk 18]
```diff
@@ -921,8 +1112,10 @@ Turn() or Shift() as there is virtually no overhead. ~N
 	else
 		to_chat(user, SPAN_NOTICE("\The [src] is empty."))
 
-
 /obj/structure/magazine_box/attackby(obj/item/W, mob/living/user)
+	if(burning)
+		to_chat(user, SPAN_DANGER("It's on fire and can explode!"))
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:1117]:
```suggestion
		to_chat(user, SPAN_DANGER("It's on fire and might explode!"))
```

File: code/modules/projectiles/ammunition.dm [hunk 19]
```diff
@@ -965,7 +1158,29 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
+/obj/structure/magazine_box/proc/apply_fire_overlay(var/will_explode = FALSE)
+	//original fire overlay is made for standard mag boxes, so they don't need additional offsetting
+	var/offset_x = 0
+	var/offset_y = 0
+
+	if(limit_per_tile == 4)	//misc boxes (mre, flares etc)
+		offset_x += 1
+		offset_y += -6
+	else if(istype(src, /obj/item/ammo_box/magazine/smg/nailgun))	//this snowflake again
+		offset_y += -2
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:1178]: same issues as earlier

File: code/modules/projectiles/ammunition.dm [hunk 20]
```diff
@@ -1003,13 +1221,17 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
+
 /obj/item/ammo_box/rounds/examine(mob/user)
 	..()
 	to_chat(user, SPAN_INFO("To refill a magazine click on the box with it in your hand. Being on [SPAN_HELPFUL("HARM")] intent will fill box from the magazine."))
 	if(bullet_amount)
 		to_chat(user, "It contains [bullet_amount] round\s.")
 	else
 		to_chat(user, "It's empty.")
+	if(burning)
+		to_chat(user, SPAN_DANGER("It's on fire and can explode!"))
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:1234]:
```suggestion
		to_chat(user, SPAN_DANGER("It's on fire and might explode!"))
```

File: code/modules/projectiles/ammunition.dm [hunk 21]
```diff
@@ -1021,6 +1243,9 @@ Turn() or Shift() as there is virtually no overhead. ~N
 	qdel(src)
 
 /obj/item/ammo_box/rounds/attackby(obj/item/I, mob/user)
+	if(burning)
+		to_chat(user, SPAN_DANGER("It's on fire and can explode!"))
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:1247]:
```suggestion
		to_chat(user, SPAN_DANGER("It's on fire and might explode!"))
```

File: code/modules/projectiles/ammunition.dm [hunk 22]
```diff
@@ -1095,13 +1320,47 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
+	if(burning)
+		return
+	burning = TRUE
+	if(can_explode)
+		var/severity = 0
+		severity = round(bullet_amount / 200)
+		//we need a lot of bullets to produce an explosion.
+
+		if(severity > 0)
+			visible_message(SPAN_WARNING("\The [src] catches on fire and ammunition starts cooking off! Its gonna blow!"))
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:1335]:
```suggestion
			visible_message(SPAN_WARNING("\The [src] catches on fire and ammunition starts cooking off! It's gonna blow!"))
```

File: code/modules/projectiles/ammunition.dm [hunk 23]
```diff
@@ -1095,13 +1320,47 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
-//explosion when using flamer procs.
-/obj/item/ammo_box/rounds/flamer_fire_act(damage, flame_cause_data)
-	switch(bullet_amount)
-		if(0) return
-		if(1 to 100) explosion(loc,  0, 0, 1, 2, , , , flame_cause_data) //blow it up.
-		else explosion(loc,  0, 0, 2, 3, , , , flame_cause_data) //blow it up HARDER
-	qdel(src)
+//---------------------FIRE HANDLING PROCS
+
+/obj/item/ammo_box/rounds/flamer_fire_act(var/damage, var/datum/cause_data/flame_cause_data)
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:1325]: get rid of this duped code

File: code/modules/projectiles/ammunition.dm [hunk 24]
```diff
@@ -1237,12 +1519,85 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
 	desc = "A box of M94 marking flare packs, to brighten up your day."
 	magazine_type = /obj/item/storage/box/m94
 	num_of_magazines = 10
+	overlay_gun_type = "_m94"
 	overlay_content = "_flares"
 
+//---------------------FIRE HANDLING PROCS
+
+//flare box has unique stuff
+/obj/item/ammo_box/magazine/misc/flares/flamer_fire_act(var/damage, var/datum/cause_data/flame_cause_data)
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:1535]: there is still some duped code in here

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/ammunition.dm [hunk 25]
```diff
@@ -398,27 +447,114 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
+				var/obj/structure/magazine_box/host_box = loc
+				host_box.apply_fire_overlay(TRUE)
+				host_box.SetLuminosity(3)
+				host_box.visible_message(SPAN_WARNING("\The [src] catches on fire and ammunition starts cooking off! It's gonna blow!"))
+			else
+				apply_fire_overlay(TRUE)
+				SetLuminosity(3)
+				visible_message(SPAN_WARNING("\The [src] catches on fire and ammunition starts cooking off! It's gonna blow!"))
+			addtimer(CALLBACK(src, .proc/explode, severity, flame_cause_data), max(5 - severity, 2))	//the more ammo inside, the faster and harder it cooks off
+			return
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:526]: all of this can easily be put into the parent

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/ammunition.dm [hunk 26]
```diff
@@ -398,27 +452,107 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
+		host_box.SetLuminosity(3)
+		host_box.visible_message(SPAN_WARNING("\The [src] catches on fire!"))
+		//need to make sure we also delete the structure box
+		addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, host_box), 5 SECONDS)
+	else
+		visible_message(SPAN_WARNING("\The [src] catches on fire!"))
+		apply_fire_overlay()
+		SetLuminosity(3)
+		addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, src), 5 SECONDS)
+	return
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:545]: this code is STILL duped in three different places

File: code/modules/projectiles/ammunition.dm [hunk 27]
```diff
@@ -1226,6 +1474,48 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
+		host_box.apply_fire_overlay()
+		host_box.SetLuminosity(3)
+		host_box.visible_message(SPAN_WARNING("\The [src] catches on fire!"))
+		addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, host_box), 7 SECONDS)
+	else
+		apply_fire_overlay()
+		SetLuminosity(3)
+		visible_message(SPAN_WARNING("\The [src] catches on fire!"))
+		addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, src), 5 SECONDS)
+	return
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:1504]: duped here

File: code/modules/projectiles/ammunition.dm [hunk 28]
```diff
@@ -1237,12 +1527,83 @@ Turn() or Shift() as there is virtually no overhead. ~N
...
+		host_box.SetLuminosity(3)
+		host_box.visible_message(SPAN_WARNING("\The [src] catches on fire!"))
+		//need to make sure we also delete the structure box
+		addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, host_box), 5 SECONDS)
+	else
+		visible_message(SPAN_WARNING("\The [src] catches on fire!"))
+		apply_fire_overlay()
+		SetLuminosity(3)
+		addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, src), 5 SECONDS)
+	return
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:1561]: and here

all this stuff can be put into a single proc

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/ammo_boxes.dm [hunk 29]
```diff
@@ -0,0 +1,1289 @@
...
+/obj/structure/magazine_box/attack_hand(mob/living/user)
+	if(burning)
+		to_chat(user, SPAN_DANGER("It's on fire and might explode!"))
+		return
+	if(item_box.contents.len)
+		if(!item_box.handfuls)
+			var/obj/item/ammo_magazine/AM = pick(item_box.contents)
+			item_box.contents -= AM
+			user.put_in_hands(AM)
+			to_chat(user, SPAN_NOTICE("You retrieve a [AM] from \the [src]."))
```
Comment (stanalbatross) [inline code/modules/projectiles/ammo_boxes.dm:793]:
```suggestion
			to_chat(user, SPAN_NOTICE("You retrieve \a [AM] from \the [src]."))
```

File: code/modules/projectiles/ammo_boxes.dm [hunk 30]
```diff
@@ -0,0 +1,1289 @@
...
+	var/obj/structure/magazine_box/host_box
+	if(istype(loc, /obj/structure/magazine_box))
+		host_box = loc
+	var/flare_amount = get_severity()
+	//need to make sure we delete the structure box if it exists, it will handle the deletion of ammo box inside
+	addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, (host_box ? host_box : src)), 7 SECONDS)
+	if(flare_amount > 0)
+		handle_side_effects(host_box, TRUE)
+		for(var/i = 1, i <= flare_amount, i++)
+			addtimer(CALLBACK(src, .proc/explode, src), rand(1, 6) SECONDS)
```
Comment (stanalbatross) [inline code/modules/projectiles/ammo_boxes.dm:1256]: it seems to just qdel before it ever does the flare thing

**Review by stanalbatross — CHANGES_REQUESTED** [review]
look into reducing the intensity of the explosions of single magazines/handfuls

otherwise fine

**Review by stanalbatross — APPROVED** [review]

**Comment by JeserTheFox** [comment]
Fixed merge conflict.

**Review by nauticall — APPROVED** [review]
sprites look good! approved

**Pull request merged**